### PR TITLE
Feature(login): oauth weixin、gitHub

### PR DESCRIPTION
**Pull request by 15168440402**

### CODE REVIEW

File: src/Contracts/Masa.Auth.Contracts.Admin/Subjects/StaffDetailDto.cs [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ﻿// Copyright (c) MASA Stack All rights reserved.
 // Licensed under the Apache License. See LICENSE.txt in the project root for license information.
 
+using Masa.BuildingBlocks.StackSdks.Auth.Contracts.Model;
```
Comment (doddgu) [inline src/Contracts/Masa.Auth.Contracts.Admin/Subjects/StaffDetailDto.cs:4]: Imports.

File: src/Contracts/Masa.Auth.Contracts.Admin/Subjects/UserDetailDto.cs [hunk 2]
```diff
@@ -1,6 +1,8 @@
 ﻿// Copyright (c) MASA Stack All rights reserved.
 // Licensed under the Apache License. See LICENSE.txt in the project root for license information.
 
+using Masa.BuildingBlocks.StackSdks.Auth.Contracts.Model;
```
Comment (doddgu) [inline src/Contracts/Masa.Auth.Contracts.Admin/Subjects/UserDetailDto.cs:4]: Imports.

File: src/Services/Masa.Auth.Service.Admin/Application/Sso/QueryHandler.cs [hunk 3]
```diff
@@ -371,5 +371,33 @@ public async Task GetCustomLoginDetailAsync(CustomLoginDetailQuery query)
...
 
+    [EventHandler]
+    public async Task CustomLoginByClientIdQueryAsync(CustomLoginByClientIdQuery query)
+    {
+        var customLogin =  await _authDbContext.Set<CustomLogin>()
+                                                .Where(customLogin => customLogin.Enabled == true)
+                                                .Include(customLogin => customLogin.ThirdPartyIdps)
+                                                .ThenInclude(thirdPartyIdp => thirdPartyIdp.ThirdPartyIdp)
+                                                .Include(customLogin => customLogin.RegisterFields)
+                                                .AsSplitQuery()
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/Application/Sso/QueryHandler.cs:382]: AsSplitQuery?

File: src/Services/Masa.Auth.Service.Admin/Program.cs [hunk 4]
```diff
@@ -144,7 +144,12 @@ await builder.Services
 });
 builder.Services.RemoveAll(typeof(IProcessor));
 
-var app = builder.Services.AddServices(builder);
+var app = builder.AddServices(options => 
+{
+    options.DisableAutoMapRoute = true;
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/Program.cs:149]: Do you want to specify it manually? If so, you need add todo comment remind remove it before v1.0.

File: src/Services/Masa.Auth.Service.Admin/appsettings.Development.json [hunk 5]
```diff
@@ -7,12 +7,12 @@
     }
   },
   "ConnectionStrings": {
-    "DefaultConnection": "Server=10.10.90.37,30100;Database=masa-auth-v2;User Id=sa;Password=p@ssw0rd;"
+    "DefaultConnection": "Server=10.10.90.37,30100;Database=masa-auth-wuweilai-ceshi;User Id=sa;Password=p@ssw0rd;"
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/appsettings.Development.json:10]: wuweilai-ceshi?

File: src/Services/Masa.Auth.Service.Admin/appsettings.Development.json [hunk 6]
```diff
@@ -7,12 +7,12 @@
...
   },
   "IsolationConnectionStrings": [
     {
       "Environment": "development",
-      "ConnectionString": "Server=10.10.90.37,30100;Database=masa-auth-v2;User Id=sa;Password=p@ssw0rd;"
+      "ConnectionString": "Server=10.10.90.37,30100;Database=masa-auth-wuweilai-ceshi;User Id=sa;Password=p@ssw0rd;"
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/appsettings.Development.json:15]: wuweilai-ceshi?

File: src/Web/Masa.Auth.Security.OAuth.Providers/Infrastructure/Identity.cs [hunk 7]
```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Auth.Security.OAuth.Providers;
+
+public class Identity
+{
+    /// <summary>
+    /// 唯一标识，一般为用户 ID
```
Comment (doddgu) [inline src/Web/Masa.Auth.Security.OAuth.Providers/Infrastructure/Identity.cs:9]: Use English.

File: src/Web/Masa.Auth.Security.OAuth.Providers/Infrastructure/UserClaims.cs [hunk 8]
```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Auth.Security.OAuth.Providers;
+
+public static class UserClaims
+{
+    [Description("唯一标识，一般为用户 ID")]
```
Comment (doddgu) [inline src/Web/Masa.Auth.Security.OAuth.Providers/Infrastructure/UserClaims.cs:8]: Add todo comment, remind for support I18n in the future.

File: src/Web/Masa.Auth.Security.OAuth.Providers/WeChat/WeChatBuilder.cs [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+public class WeChatBuilder : IIdentityBuilder, ILocalAuthenticationDefaultBuilder, IAuthenticationInject, IAuthenticationInstanceBuilder
+{
+    public string Scheme { get; } = "WeChat";
+
+    public AuthenticationDefaults AuthenticationDefaults { get; } = new AuthenticationDefaults
+    {
+        HandlerType = typeof(WeChatAuthenticationHandler),
+        Scheme = "WeChat",
+        DisplayName = WeixinAuthenticationDefaults.DisplayName,
+        Icon = "https://masa-cdn-dev.oss-cn-hangzhou.aliyuncs.com/wechat.ico",
```
Comment (doddgu) [inline src/Web/Masa.Auth.Security.OAuth.Providers/WeChat/WeChatBuilder.cs:17]: use cdn.

File: src/Web/Masa.Auth.Web.Sso/Pages/Account/Login/Index.razor [hunk 10]
```diff
@@ -40,7 +40,7 @@
     </MCard>
 }
 
-@if (_viewModel.VisibleExternalProviders.Any())
+@*@if (_viewModel.VisibleExternalProviders.Any())
```
Comment (doddgu) [inline src/Web/Masa.Auth.Web.Sso/Pages/Account/Login/Index.razor:43]: ?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Web/Masa.Auth.Security.OAuth.Providers/WeChat/WeChatBuilder.cs (hunk 9 above)
Comment (15168440402) [reply src/Web/Masa.Auth.Security.OAuth.Providers/WeChat/WeChatBuilder.cs:17]: image
这就是cdn地址吧

Comment (doddgu) [reply src/Web/Masa.Auth.Security.OAuth.Providers/WeChat/WeChatBuilder.cs:17]: 不是

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**